_ingest-pipelines/processors/user-agent.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: default
+title: User agent
+parent: Ingest processors
+nav_order: 330
+---
+
+# User agent processor
+
+The `user_agent` processor is used to extract information from the user agent string, such as the browser, device, and operating system used by the client.
+
+The following is the syntax for the `user_agent` processor:
+
+```json
+{
+  "processor": {
+    "user_agent": {
+      "field": "user_agent",
+      "target_field": "user_agent_info"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `user_agent` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The field containing the user agent string.
+`target_field` | Optional | The field to store the extracted user agent information. If not specified, the information is stored in the `user_agent` field.
+
+
```

Comment:
@@gaobinlong Is this parameter applicable here?

Reply:
`override_target ` should be removed, and the following 2 parameters should be added:
```
`regex_file `  | Optional  | A file containing some regex expressions which can be used to parse the user agent string, this file must be under the directory of `config/ingest-user-agent` in the OpenSearch package, the default file is [regexes.yaml](https://github.com/opensearch-project/OpenSearch/blob/main/modules/ingest-user-agent/src/main/resources/regexes.yml) which is shipped with OpenSearch. |
`properties `  | Optional  | What properties should be added to the `target_field`, defaults to ["name", "major", "minor", "patch", "build", "os", "os_name", "os_major", "os_minor", "device"].  |
```